cfdm/test/test_read_write.py

```diff
@@ -886,6 +886,29 @@ def test_read_subsampled_coordinates(self):
             np.allclose(rlon, olon.data.array, atol=2.2e-05, rtol=0)
         )
 
+    def test_read_original_filenames(self):
+        """Test the setting of original file names."""
+        f = cfdm.read(self.filename)[0]
+        x = f.dimension_coordinate("grid_longitude")
+
+        self.assertEqual(len(f.original_filenames()), 1)
+        self.assertEqual(len(x.original_filenames()), 1)
+
+        self.assertEqual(len(x.original_filenames(clear=True)), 1)
+        self.assertEqual(x.original_filenames(), set())
+
+        self.assertEqual(len(f.original_filenames(clear=True)), 1)
+        self.assertEqual(f.original_filenames(), set())
```

Comment:
Since we're just running the same assertions here for `x` and `f`, might be good to consolidate into a loop?

Reply:
Good idea. Done in https://github.com/NCAS-CMS/cfdm/pull/216/commits/d59240610d4538c3ab297dd58975b44b744a948f